keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
+  AuthenticationType AuthenticationType
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string
 }
 ```
 
-## ProvisonerDeleteBucket
+## BucketInfo
 
-This call is made to delete the bucket in the backend. If the bucket has already been deleted, then no error should be returned. The call to *ProvisonerDeleteBucket* MUST be idempotent.
+Resource mounted into pods containing information for applications to gain access to buckets.
 
-```
-message ProvisionerDeleteBucketRequest {
-    // This field is REQUIRED
-    string bucket_name = 1;
+```yaml
+BucketInfo {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketInfoSpec {
+    // BucketName is the name of the Bucket 
+    BucketName string
+
+    // AuthenticationType denotes the style of authentication
+    // It can be one of
+    // KEY - access, secret tokens based authentication
+    // IAM - implicit authentication of pods to the OSP based on service account mappings
+    AuthenticationType AuthenticationType
+
+    // Endpoint is the URL at which the bucket can be accessed
+    Endpoint string
     
-    map<string,string> bucket_context = 2;    
-}
-
-message ProvisionerDeleteBucketResponse {
-     // Intentionally left blank
+    // Region is the vendor-defined region where the bucket "resides"
+    Region string
+    
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
+  }
 }
 ```
 
-## ProvisionerGrantBucketAccess
+# COSI Driver
+
+A component that runs alongside COSI Sidecar and satisfies the COSI gRPC API specification. Sidecar and driver work together to orchestrate changes in the OSP. The driver acts as a gRPC server to the COSI Sidecar. Each COSI driver is identified by a unique id.
 
-This call grants access to a particular principal. The principal is the account for which this access should be granted. 
+The sidecar uses the unique id to direct requests to the appropriate drivers. Multiple instances of drivers with the same id will be added into a group and only one of them will act as the leader at any given time.
 
-If the principal is set, then it should be used as the username of the created credentials or in some way should be deterministically used to generate a new credential for this request. This principal will be used as the unique identifier for deleting this access by calling ProvisionerRevokeBucketAccess
+## COSI Driver gRPC API
 
-If the `principal` is empty, then a new service account should be created in the backend that satisfies the requested `access_policy`. The username/principal for this service account should be set in the `principal` field of `ProvisionerGrantBucketAccessResponse`.
+#### ProvisionerGetInfo
```

Comment:
Suggest `DriverGetInfo` because the driver does more than provisioning.

Reply:
If it is only a cosmetic change, I can make it after all other parts are given the green light. It would give me enough time to bring up the name change in the community and get their feedback. I suspect nobody will have a problem with calling it driver, but it would be good to check-in anyways.